picotest_macros/src/lib.rs

```diff
@@ -4,21 +4,44 @@ use darling::ast::NestedMeta;
 use darling::{Error, FromMeta};
 use proc_macro::TokenStream;
 use quote::{format_ident, quote};
-use syn::{parse_macro_input, parse_quote, Attribute, Item, Stmt};
-use utils::traverse_use_item;
+use syn::{parse_macro_input, parse_quote, Item};
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+enum Scope {
+    Session,
+    Module,
+}
+
+impl From<String> for Scope {
+    fn from(value: String) -> Self {
+        match value.to_lowercase().as_str() {
+            "session" => Scope::Session,
+            "module" => Scope::Module,
+            _ => {
+                eprintln!("Undefined scope: '{value}'. Use session scope");
+                Scope::Session
+            }
+        }
+    }
+}
 
 fn plugin_path_default() -> String {
     ".".to_string()
 }
 fn plugin_timeout_secs_default() -> u64 {
     5
 }
+fn default_scope() -> String {
+    "session".into()
+}
 #[derive(Debug, FromMeta)]
 struct PluginCfg {
     #[darling(default = "plugin_path_default")]
     path: String,
     #[darling(default = "plugin_timeout_secs_default")]
     timeout: u64,
+    #[darling(default = "default_scope")]
```

Comment:
Можно добавить метод `Scope::default` и возвращать `Scope::Session` по умолчанию:
```rust
impl Default for Scope {
    fn default() -> Self {
        Scope::Session
    }
}
```
А здесь напрямую указать этот метод:
```suggestion
    #[darling(default = "Scope::default")]
    scope: Scope
```
Чтобы darling понимал как это парсить, можно определить трейт FromMeta для Scope:
```rust
impl FromMeta for Scope {
    fn from_string(value: &str) -> Result<Self, Error> {
        Ok(value.into())
    }
}
```

Ну и в самом трейте `impl From<String> for Scope` тоже можно в случае ненайденного значения возвращать `Scope::default()`